packages/node-core/src/indexer/sandbox.ts

```diff
@@ -46,11 +51,65 @@ export class Sandbox extends NodeVM {
         },
       })
     );
+    this.root = option.root;
+
+    const filename = 'dist/index.js';
+    const sourceMapPath = path.join(this.root, filename);
+
+    if (existsSync(sourceMapPath)) {
+      this.sourceMap = this.decodeSourceMap(sourceMapPath);
+    }
   }
 
   async runTimeout<T = unknown>(duration: number): Promise<T> {
     return timeout(this.run(this.script), duration);
   }
+
+  async convertStack(stackTrace: any) {
+    if (!this.sourceMap) {
+      logger.warn('Unable to find a source map. Rebuild your project with latest @subql/cli to generate a source map.');
+      logger.warn('Logging unresolved stack trace.');
+      return stackTrace;
+    }
+    const regex = /index\.js:([0-9]+):([0-9]+)/gi;
+    const matches = [...stackTrace.matchAll(regex)];
+
+    for (const match of matches) {
+      const lineNumber = Number.parseInt(match[1]);
+      const columnNumber = Number.parseInt(match[2]);
+      const lineInfo = await this.findLineInfo(this.sourceMap, lineNumber, columnNumber);
+      const newLineTrace = `${lineInfo.source}:${lineInfo.line}:${lineInfo.column}`;
+      const filepath = path.join(this.root, 'dist/index.js');
+      stackTrace = stackTrace.replace(`${filepath}:${match[1]}:${match[2]}`, newLineTrace);
+    }
+
+    return stackTrace;
+  }
+
+  decodeSourceMap(sourceMapPath: string) {
+    const source = readFileSync(sourceMapPath).toString();
+    const sourceMapBase64 = source.split(`//# sourceMappingURL=data:application/json;charset=utf-8;base64,`)[1];
+    if (!sourceMapBase64) {
+      logger.warn('Unable to find a source map for project');
+      logger.warn('Build your project with latest @subql/cli to generate a source map');
+      return;
+    }
+    const sourceMap = Buffer.from(sourceMapBase64, 'base64').toString();
+    const json = JSON.parse(sourceMap);
+
+    return json;
+  }
+
+  async findLineInfo(
+    sourcemap: any,
+    compiledLineNumber: number,
+    compiledColumnNumber: number
+  ): Promise<NullableMappedPosition> {
+    const consumer = await new SourceMapConsumer(sourcemap);
```

Comment:
```suggestion
    const consumer = new SourceMapConsumer(sourcemap);
```

Reply:
the constructor seems to return a promise: https://github.com/mozilla/source-map/blob/4e304db915cecd42505feb007766a034395c6eb7/lib/source-map-consumer.js#L18